packages/xo-web/src/common/utils.js

```diff
@@ -371,13 +369,23 @@ export const resolveResourceSets = resourceSets =>
 //   id: 42,
 // }, 32)
 //
-// newString === 'foo_COPY_foo_42_32'
+// newString === 'foo_COPY_foo_42_32%'
 // ```
 export function buildTemplate(pattern, rules) {
-  const regExp = new RegExp(join(map(keys(rules), escapeRegExp), '|'), 'g')
+  const syntaxesToReplace = []
+  Object.keys(rules).forEach(syntax =>
+    syntaxesToReplace.push(escapeRegExp(syntax), escapeRegExp(`\\${syntax}`))
```

Comment:
Are you sure it won't match `%` before `\%` thus replacing `\%` with `\42`?

Reply:
Yes, i m sure.